src/components/localtunneldebugger/localtunneldebugger.ts

```diff
@@ -0,0 +1,46 @@
+import * as vscode from 'vscode';
+
+import { LocalTunnelDebugProvider } from './localtunneldebugger.extension';
+import { getLocalTunnelDebugProvider } from '../config/config';
+
+export class LocalTunnelDebugger {
+    private readonly providers = Array.of<LocalTunnelDebugProvider>();
+
+    register(provider: LocalTunnelDebugProvider): void {
+        console.log(`Registered local tunnel debugger type ${provider.id}`);
+        this.providers.push(provider);
+    }
+
+    startLocalTunnelDebugProvider(target?: any): void {
+        const browseExtensions = "Find Providers on Marketplace";
+        if (this.providers.length === 0)
+        {
+            vscode.window.showInformationMessage('You do not have a Local Tunnel Debug Provider installed.', browseExtensions)
+            .then((selection: string | undefined) => {
+                if (selection === browseExtensions) {
+                    vscode.commands.executeCommand('extension.vsKubernetesFindLocalTunnelDebugProviders');
+                }
+            });
+            return;
+        }
+
+        let providerName: string = getLocalTunnelDebugProvider();
+        if (providerName === "") {
```

Comment:
Is it possible that this `providerName` is `undefined`, if the user manually remove this field from the settings? Or do we have the guarantee that it will always be set to an empty string if missing?

Reply:
Hmm, good question. I did set the default value in package.json, and from the [docs](https://code.visualstudio.com/api/references/contribution-points#contributes.configuration) it looks like even if they remove it from settings.json the fallback value should get resolved? However, this is an easy case to handle so I'll updating to include the `undefined` case just in case. I see some of the other configurations in this file have varying ways of handling this